contexts/DonationContext/src/UseCases/AddDonation/AddDonationUseCase.php

```diff
@@ -64,6 +64,10 @@ public function addDonation( AddDonationRequest $donationRequest ): AddDonationR
 			$donation->markForModeration();
 		}
 
+		if ( $this->policyValidator->isAutoDeleted( $donationRequest ) ) {
+			$donation->markAsDeleted();
```

Comment:
You're totally right. What I said only makes sense for memberships, for donations we can't track moderation and deletion separately.